src/utils/allocator_n64.h

```diff
@@ -0,0 +1,65 @@
+#ifndef ALLOCATOR_DEFAULT_H
+#define ALLOCATOR_DEFAULT_H
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+// Allocator options
+#define ALLOC_HINT_TEXTURE 0x00000001
+
+#define omf_free(ptr)                                                                                                  \
+    do {                                                                                                               \
+        free(ptr);                                                                                                     \
+        (ptr) = NULL;                                                                                                  \
+    } while(0)
+
+static inline void *omf_malloc_real(size_t size, const char *file, int line) {
+    void *ret = malloc(size);
+    if(ret != NULL)
+        return ret;
+    fprintf(stderr, _text_malloc_error, size, file, line);
+    abort();
+}
+
+static inline void *omf_calloc_real(size_t nmemb, size_t size, const char *file, int line) {
+    void *ret = calloc(nmemb, size);
+    if(ret != NULL)
+        return ret;
+    fprintf(stderr, _text_calloc_error, nmemb, size, file, line);
+    abort();
+}
+
+static inline void *omf_realloc_real(void *ptr, size_t size, const char *file, int line) {
+    void *ret = realloc(ptr, size);
+    if(ret != NULL)
+        return ret;
+    fprintf(stderr, _text_realloc_error, ptr, size, file, line);
+    abort();
+}
+
+static inline void *omf_alloc_with_options_real(size_t nmemb, size_t size, int options, const char *file, int line) {
+    void *ret = NULL;
+    if((options & ALLOC_HINT_TEXTURE) != 0) {
+        assertf(size != 0, "texture size was 0");
+        ret = malloc_uncached_aligned(64, size);
```

Comment:
malloc(0) is generally valid, although not a great use of CPU time.
if libdragon's malloc_uncached_aligned explodes when size is 0, perhaps we should return NULL here?
if malloc(0) is not a bug, perhaps a warning is more fitting than an assert? both compile to nothing in release builds.

Reply:
just noticed allocator.h declares hardly anything
this is unfortunate and not this PR's fault

I'd prefer if allocator.h defined all the symbols that users of allocator.h are supposed to use: `omf_free`, etc..
`omf_realloc_real` and similar (omf_free_real) can continue to be only called in allocator.h and only defined in allocator_platform.h, as they're not public symbols.

Reply:
> if libdragon's malloc_uncached_aligned explodes when size is 0, perhaps we should return NULL here?

It does blow up with size 0, but I am adamant that we should not be allocating textures of size 0 and would like those to be caught and fixed.